includes/managers/icons.php

```diff
@@ -377,25 +385,67 @@ public static function render_icon( $icon, $attributes = [], $tag = 'i' ) {
 	 *
 	 * @return array
 	 */
-	public static function fa4_to_fa5_value_migration( $value ) {
-		static $migration_dictionary = false;
-		if ( '' === $value ) {
-			return [
-				'value' => '',
+	public static function fa_icon_value_migration( $value ) {
+		static $migration_dictionary = [];
+
+		$mapping_files = self::get_mapping_files();
+		$found_in_mapping = false;
+
+		if ( is_string( $value ) ) {
+			if ( '' === $value ) {
+				return [
+					'value' => '',
+					'library' => '',
+				];
+			}
+
+			$value = [
+				'value' => $value,
 				'library' => '',
 			];
 		}
-		if ( false === $migration_dictionary ) {
-			$migration_dictionary = json_decode( Utils::file_get_contents( ELEMENTOR_ASSETS_PATH . 'lib/font-awesome/migration/mapping.js' ), true );
+
+		foreach ( $mapping_files as $mapping_file ) {
+			if ( ! isset( $migration_dictionary[ $mapping_file ] ) ) {
+				$migration_dictionary[ $mapping_file ] = json_decode( Utils::file_get_contents( $mapping_file ), true );
+			}
+
+			if ( isset( $migration_dictionary[ $mapping_file ][ $value['value'] ] ) ) {
+				$value = $migration_dictionary[ $mapping_file ][ $value ]['value'];
+				$found_in_mapping = true;
+			}
 		}
-		if ( isset( $migration_dictionary[ $value ] ) ) {
-			return $migration_dictionary[ $value ];
+
+		// Make sure there's no value in the old 'fa ' format, even if no replacement found.
+		if ( ! $found_in_mapping && '' === $value['library'] ) {
+			$value = [
+				'value'   => 'fas ' . str_replace( 'fa ', '', $value ),
+				'library' => 'fa-solid',
+			];
 		}
 
-		return [
-			'value' => 'fas ' . str_replace( 'fa ', '', $value ),
-			'library' => 'fa-solid',
-		];
+		return $value;
+	}
+
+	/**
+	 * Get the mapping files for the migration.
+	 *
+	 * @return array
+	 */
+	public static function get_mapping_files() {
+		$current_version = self::get_current_fa_version();
+		$mapping_files = [];
+
+		for ( $i = $current_version - 1; $i >= 1; $i-- ) {
+			$mapping_by_version = sprintf( 'mapping-v%s-to-v%s', $i, $i + 1 );
+			$mapping_file = ELEMENTOR_ASSETS_PATH . "/lib/font-awesome/migration/$mapping_by_version.js";
+
+			if ( file_exists( $mapping_file ) ) {
```

Comment:
+ Maybe instead of checking if the files exist for each icon, consider consolidating this method with the iteration in lines 409-411, maybe this method can return the "cached" decoded json,

Reply:
Yeah, you are right. I did not consolidate, but changed the code a bit now.